### PR TITLE
CA-359965: Consider `physical_utilisation` value when checking for free space in thinly provisioned SRs

### PR DESCRIPTION
**Pull request by danilo-delbusso**
`VdiCreationCanProceed` is only used for this use case and when a new VM is being created. In the latter case, we don't really care about physical utilisation of VMs, since it's only checking template disk sizes.

### CODE REVIEW

**Comment by danilo-delbusso** [comment]
blocked by partial freezing of `master`, this is not going until we unblock the rest

**Review by AaronRobson — APPROVED** [review]

File: XenModel/XenAPI-Extensions/SR.cs
Comment (CitrixChris) [inline XenModel/XenAPI-Extensions/SR.cs:520]: above you use one line if statements then here you use brackets, stick to the one line to keep it consistent

**Review by CitrixChris — CHANGES_REQUESTED** [review]

**Review by CitrixChris — APPROVED** [review]

**Comment by kc284** [comment]
I have added some extra refactoring in https://github.com/xenserver/xenadmin/pull/3028.

**Review by kc284 — APPROVED** [review]

**Review by CitrixChris — APPROVED** [review]

**Review by CitrixChris — APPROVED** [review]

**Pull request merged**
